datastore/google/cloud/datastore/_gax.py

```diff
@@ -90,20 +91,20 @@ class _DatastoreAPIOverGRPC(object):
     :type connection: :class:`Connection`
     :param connection: A connection object that contains helpful
                        information for making requests.
-
-    :type secure: bool
-    :param secure: Flag indicating if a secure stub connection is needed.
     """
 
-    def __init__(self, connection, secure):
-        if secure:
+    def __init__(self, connection):
+        parse_result = six.moves.urllib_parse.urlparse(
+            connection.api_base_url)
+        host = parse_result.hostname
```

Comment:
The parsing would move to `Client.__init__` if it moved (I want `Connection` to be a dumb proxy). I just thought it was overkill to carry around two values (the host / URL and the `secure` flag)